src/main/java/com/cinema/config/validations/PasswordMatchValidator.java

```diff
@@ -0,0 +1,27 @@
+package com.cinema.config.validations;
+
+import com.cinema.model.dto.UserRequestDto;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class PasswordMatchValidator
+        implements ConstraintValidator<PasswordMatchConstraint, UserRequestDto> {
+    private String password;
+    private String repeatPassword;
+
+    public void initialize(PasswordMatchConstraint constraint) {
+        this.password = constraint.password();
+        this.repeatPassword = constraint.repeatPassword();
+    }
+
+    public boolean isValid(UserRequestDto userRequestDto, ConstraintValidatorContext context) {
+
+        String passwordValue = (String) new BeanWrapperImpl(userRequestDto)
+                .getPropertyValue(password);
+        String repeatPasswordValue = (String) new BeanWrapperImpl(userRequestDto)
+                .getPropertyValue(repeatPassword);
+
+        return passwordValue == null || passwordValue.equals(repeatPasswordValue);
```

Comment:
what if password is null, and repeatPassword is not null?

Reply:
I have NotNull on this field in the Dto - so I have a response "password can't be null". A message that those two "are not equal" will not appear but in any case we have an error which should be fixed.